sync/commit.py

```diff
@@ -392,19 +400,22 @@ def _apply_patch(patch,  # type: str
                     (stdout, stderr) = proc.communicate(patch)
                     if not proc.returncode == 0:
                         err_msg = ("%s\n\nPatch failed (status %i):\nstdout:\n%s\nstderr:\n%s" %
-                                   (err_msg, proc.returncode, stdout, stderr))
+                                   (err_msg,
+                                    proc.returncode,
+                                    stdout.decode("utf8", "replace") if stdout else "",
+                                    stderr.decode("utf8", "replace") if stderr else ""))
                     else:
                         err_msg = None
-                        prefix = "+++ "
+                        prefix = b"+++ "
                         paths = []
                         for line in patch.splitlines():
                             if line.startswith(prefix):
-                                path = "%s/%s" % (
-                                    dest_prefix,
-                                    "/".join(line[len(prefix):].split("/")[strip_dirs:]))
+                                path_parts_bytes = line[len(prefix):].split(b"/")[strip_dirs:]
```

Comment:
I'm a bit wary of adding these changes into this PR

Reply:
yes without tests it's always hard to make a decision on this kind of things. and the functions being big too it makes it even more complicated.

Reply:
To be fair this code is exercised by tests since we have quite a few integration-type tests. But it still seems a little risky to add additional cleanup that isn't really the main point of the patch.